docs/dev/building-on-zksync/contracts/differences-with-ethereum.md

```diff
@@ -0,0 +1,192 @@
+} 
+```
+
+The following code will not work, because the compiler will not be aware of the bytecode beforehand:
+
+```solidity
+function myFactory(bytes memory bytecode) public {
+   assembly {
+      addr := create(0, add(bytecode, 0x20), mload(bytecode))
+   }
+}
+```
+
+Unfortunately it is not possible to distinguish between the cases above in compile time. And so it *highly recommended* to include tests for your factory regarding deploying the child contracts. 
+
+Also, note that for zkEVM bytecode we contain the different address derivation from Ethereum. The exact formulas are available in our SDK:
+
+```typescript
+export function create2Address(sender: Address, bytecodeHash: BytesLike, salt: BytesLike, input: BytesLike) {
+    const prefix = ethers.utils.keccak256(ethers.utils.toUtf8Bytes('zksyncCreate2'));
+    const inputHash = ethers.utils.keccak256(input);
+    const addressBytes = ethers.utils
+        .keccak256(ethers.utils.concat([prefix, ethers.utils.zeroPad(sender, 32), salt, bytecodeHash, inputHash]))
+        .slice(26);
+    return ethers.utils.getAddress(addressBytes);
+}
+
+export function createAddress(sender: Address, senderNonce: BigNumberish) {
+    const prefix = ethers.utils.keccak256(ethers.utils.toUtf8Bytes('zksyncCreate'));
+    const addressBytes = ethers.utils
+        .keccak256(
+            ethers.utils.concat([
+                prefix,
+                ethers.utils.zeroPad(sender, 32),
+                ethers.utils.zeroPad(ethers.utils.hexlify(senderNonce), 32)
+            ])
+        )
+        .slice(26);
+
+    return ethers.utils.getAddress(addressBytes);
+}
+```
+
+### Differences in `block.timestamp` and `block.number`
+These variables behave differently from L1. Read more [about blocks in zkSync](../../developer-guides/transactions/blocks.md#blocks-in-zksync-era).
+
+### Differences in `COINBASE` `DIFFICULTY` `BASEFEE` 
+- `COINBASE` returns our Bootloader contract address, `0x0000000000000000000000000000000000008001` 
+- `DIFFICULTY` returns a constant value of `2500000000000000` 
+- `BASEFEE` is not a constant, but is defined by the fee model. Most of the time it is 0.25 gwei, but under very high L1 gas prices it may rise.
+
+### Use of `codesize`/`codecopy` and `calldata` in the deploy code
+`Calldata` in the constructor is not empty as on the EVM, on zkEVM calldata contains constructor arguments.
+`codesize` has the same behavior as `calldatasize`, `codecopy` as `calldatacopy`.
+To make it works datasize of the current object is 0.
+
+
+## Opcode Differences
+
+### Removed Opcodes
+
+| Opcode      | Explanation |
+| ----------- | ----------- |
+| `SELFDESTRUCT`      | Considered harmful and deprecated in [EIP-6049](https://eips.ethereum.org/EIPS/eip-6049)       |
+| `CALLCODE`   | Deprecated in [EIP-2488](https://eips.ethereum.org/EIPS/eip-2488) in favor of `DELEGATECALL`        |
+| `EXTCODECOPY` |  |
+| `CODECOPY` | Replaced with `CALLDATACOPY` |
```

Comment:
It is replaced only in the deploy code. In the runtime code, It's forbidden. Also, I think it makes sense to mention here `CODESIZE`(it is replaced with `calldatasize` in the deploy code). Detailed info: 
- `CODECOPY` is not supported in the runtime, for the yul pipeline it will cause a compilation error, for the EVMLA it will be replaced with a no-op.
- `CODESIZE` - everything is the same as for `CODECOPY`, but will return 0 instead no-op.